libass/ass_render.h

```diff
@@ -339,6 +343,21 @@ struct ass_renderer {
     const BitmapEngine *engine;
 
     ASS_Style user_override_style;
+
+#if ENABLE_THREADS
+    ASSMutex mutex;
+    int mutex_inited;
```

Comment:
This bunch of boolean flags isn't really necessary outside of `ass_renderer_init()`. I think it's better to use local `bool` variables there and use `thread_start_failed` as collective flag for mutexes/condvars init status.

Reply:
I need to keep track of "we tried to set up threads, but failed, so we should stop trying" separately from "this individual object needs to be deleted later".

Reply:
But I don't see the need to keep all the `mutex`, `main_cond`, `pool_cond` alive till `ass_renderer_done()` in case of failure.